src/app/view/components/_common/site-selector/site-selector.component.ts

```diff
@@ -85,6 +86,14 @@ export class SiteSelectorComponent implements OnInit, OnChanges {
      */
     handleSitesRefresh(): void {
         this.paginationService.getCurrentPage().subscribe((items) => {
+            this.sites = items.splice(0);
+            this.getSitesList();
+
+            // if current site not being listed on updated sites list (socket message) then switch to default site
```

Comment:
I think we really need is a TODO instead, some thing like: 'look for a better implementation...' (we need a refactoring into site.service)

Reply:
This will not work if the list have more that one page.